### PR TITLE
Fix promotion's sales channel restriction

### PR DESCRIPTION
**Pull request by vlaux**
#### What is the purpose of this pull request?
As title says.

#### What problem is this solving?
Once the promotions form was mounted, the sales channel update received via props was ignored, making it seem like there was no sales channel restriction.

#### How should this be manually tested?
Opening [this promotion](https://pricingqa.myvtex.com/admin/promotions/1a18dd71-26ae-428b-a425-af6d816af784/) on the current version will show no SC restriction.
The same promotion now [shows it as expected](https://vlauxbeta--pricingqa.myvtex.com/admin/promotions/1a18dd71-26ae-428b-a425-af6d816af784/).

#### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/Promotion/PromotionForm.js [hunk 1]
```diff
@@ -54,6 +54,26 @@ class PromotionForm extends Component {
     }
   }
 
+  static getDerivedStateFromProps(props, state) {
```
Comment (ericreis) [inline react/components/Promotion/PromotionForm.js]: Why is this needed?

Comment (vlaux) [reply react/components/Promotion/PromotionForm.js]: Because `promotion` is [stored on the `state`](https://github.com/vtex/admin-promotions/blob/fix/ch32918/trade-policies/react/components/Promotion/PromotionForm.js#L42)
However, the promotion object is built [here](https://github.com/vtex/admin-promotions/blob/fix/ch32918/trade-policies/react/PromotionPage.js#L19), and if you take a look on the [build function](https://github.com/vtex/admin-promotions/blob/fix/ch32918/trade-policies/react/utils/promotion/index.js#L163), you might note that it depends on the `salesChannels` received via param. On the first load, they are usually not loaded yet, but a second run of this method is made when they finish loading. Anyway, since the promotion is already set on the PromotionForm state, it doesnt get updated, and thats why the `getDerivedStateFromProps` (following [react docs](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops),
> This method exists for rare use cases where the state depends on changes in props over time.

I don't like this thing at all, but it seems to me to be the most feasible way to solve this important bug before we get the chance of refactoring this whole component. 😞

**Comment by vlaux** [comment]
⚠️ There is a bug when creating a new promotion. Gonna fix it.

**Comment by ericreis** [comment]
As discussed, we thought of a better way to it and will open a new PR.
